openwisp_notifications/handlers.py

```diff
@@ -316,3 +317,19 @@ def schedule_object_notification_deletion(instance, created, **kwargs):
         tasks.delete_ignore_object_notification.apply_async(
             (instance.pk,), eta=instance.valid_till
         )
+
+
+def register_notification_observation(model, signal):
```

Comment:
I'd call this `register_notification_cache_update` or anything similar which makes it evident that by passing a model and a signal, the notification cache will be updated when that signal from that model is emitted.